cmake/external/nanopb.patch

```diff
@@ -1,15 +1,19 @@
 diff -Naur nanopb/CMakeLists.txt nanopb-fix/CMakeLists.txt
 --- nanopb/CMakeLists.txt	2021-03-22 08:50:07.000000000 -0400
 +++ nanopb-fix/CMakeLists.txt	2022-06-24 16:17:09.130783413 -0400
-@@ -41,7 +41,7 @@
+@@ -41,10 +41,10 @@
  if(nanopb_BUILD_GENERATOR)
      set(generator_protos nanopb plugin)
  
 -    find_package(PythonInterp 2.7 REQUIRED)
 +    find_package(PythonInterp 3.7 REQUIRED)
      execute_process(
          COMMAND ${PYTHON_EXECUTABLE} -c
-             "from distutils import sysconfig; print(sysconfig.get_python_lib(prefix=''))"
+-            "from distutils import sysconfig; print(sysconfig.get_python_lib(prefix=''))"
++            "import os.path, sys, sysconfig; print(os.path.relpath(sysconfig.get_path('purelib'), start=sys.prefix))"
+         OUTPUT_VARIABLE PYTHON_INSTDIR
+         OUTPUT_STRIP_TRAILING_WHITESPACE
+     )
```

Comment:
what is the significance of this `execute_process`? is it just printing something? should we just remove it?

Reply:
it was pointed out that PYTHON_INSTDIR is used [here](https://github.com/nanopb/nanopb/blob/7ee9ef9f627d85cbe1b8c4f49a3ed26eed216c77/CMakeLists.txt#L63).

Reply:
`PYTHON_INSTDIR` (the variable to which the output of the process is stored) is used below in this line:

```
DESTINATION ${PYTHON_INSTDIR}
```

https://github.com/nanopb/nanopb/blob/7ee9ef9f627d85cbe1b8c4f49a3ed26eed216c77/CMakeLists.txt#L63